### PR TITLE
use minimal entity screen for case lists

### PR DESCRIPTION
**Pull request by calellowitz**
This is the formplayer half of https://github.com/dimagi/commcare-core/pull/937
Opening for review, but it cannot be merged until that is merged and the commcare-core reference in the formplayer repo is updated.

### CODE REVIEW

File: src/main/java/org/commcare/formplayer/services/MenuSessionRunnerService.java [hunk 1]
```diff
@@ -185,15 +186,19 @@ public BaseResponseBean advanceSessionWithSelections(MenuSession menuSession,
         NotificationMessage notificationMessage = null;
         for (int i = 1; i <= selections.length; i++) {
             String selection = selections[i - 1];
-            boolean gotNextScreen = menuSession.handleInput(selection);
+
+            // minimal entity screens are only safe if there will be no further selection
```
Comment (ctsims) [inline src/main/java/org/commcare/formplayer/services/MenuSessionRunnerService.java:190]: I understand why we would need to keep the screen for a case detail, but what accounts for the 'no further selection' limitation?

**Review by ctsims — COMMENTED** [review]
Left one question for you on the implementation structure.

File: src/main/java/org/commcare/formplayer/services/MenuSessionRunnerService.java (hunk 1 above)
Comment (calellowitz) [reply src/main/java/org/commcare/formplayer/services/MenuSessionRunnerService.java:190]: To the extent I can tell, app navigation works by feeding selections as input to CLI screens (thats the loop this line is in). So for anything where a case is part of the input (like when entering a form), the `EntityScreen` needs to be able to handle case selection, and at least currently that requires the two components that I skipped in the minimal version.

Comment (ctsims) [reply src/main/java/org/commcare/formplayer/services/MenuSessionRunnerService.java:190]: ah, I see. This is about the future requests that are 'playing back' the user choices that we talked about as one of the other potential targets for performance improvement. That does sound right. Thanks for explaining

**Review by ctsims — APPROVED** [review]

**Pull request merged**